ibis/backends/exasol/tests/conftest.py

```diff
@@ -0,0 +1,334 @@
+        return f"{self.help_text} (default: {self.default})."
+
+
+class OptionGroup:
+    """
+    Wraps a set of pytest options.
+    """
+
+    def __init__(self, prefix, options):
+        self._prefix = prefix
+        self._options = tuple(Option(prefix=prefix, **kwargs) for kwargs in options)
+        self._default = {o.name: o.default for o in self._options}
+        self._env = {}
+        self._cli = {}
+        self._kwargs = ChainMap(self._cli, self._env, self._default)
+
+    @property
+    def prefix(self):
+        """The option group prefix."""
+        return self._prefix
+
+    @property
+    def options(self):
+        """A tuple of all options which are part of this group."""
+        return self._options
+
+    def kwargs(self, environment, cli_arguments):
+        """
+        Given the default values, the passed environment and cli arguments it will
+        take care of the prioritization for the option values in regard of their
+        source(s) and return a kwargs dictionary with all options and their
+        appropriate value.
+        """
+        env = {
+            o.name: o.type(environment[o.env])
+            for o in self._options
+            if o.env in environment
+        }
+        cli = {
+            o.name: getattr(cli_arguments, o.pytest)
+            for o in self.options
+            if hasattr(cli_arguments, o.pytest)
+            and getattr(cli_arguments, o.pytest) is not None
+        }
+        self._env.update(env)
+        self._cli.update(cli)
+        return self._kwargs
+
+
+@dataclass
+class Exasol:
+    """Exasol database configuration"""
+
+    host: str
+    port: int
+    username: str
+    password: str
+    database: str
+
+
+EXASOL_OPTION_GROUP = OptionGroup(
+    prefix="exasol",
+    options=(
+        {
+            "name": "host",
+            "type": str,
+            "default": "localhost",
+            "help_text": "Host where the database is running on",
+        },
+        {
+            "name": "port",
+            "type": int,
+            "default": 8563,
+            "help_text": "Port the database is listening on",
+        },
+        {
+            "name": "username",
+            "type": str,
+            "default": "sys",
+            "help_text": "Username used to authenticate against the database",
+        },
+        {
+            "name": "password",
+            "type": str,
+            "default": "exasol",
+            "help_text": "Password used to authenticate against the database",
+        },
+        {
+            "name": "database",
+            "type": str,
+            "default": "EXASOL",
+            "help_text": "Database/Schema which shall be opened",
+        },
+    ),
+)
+
+EXASOL = None
+
+
+def pytest_configure(config):
+    global EXASOL
+    cli_arguments = config.option
+    kwargs = EXASOL_OPTION_GROUP.kwargs(os.environ, cli_arguments)
+    EXASOL = Exasol(**kwargs)
+
+
+def pytest_addoption(parser):
```

Comment:
Can you give a short description of why this was needed? Not opposed to it, but we don't have special pytest options for any other backend. Just wondering if this is necessary to make testing exasol easier.

Reply:
First up: after thinking about it, I don't think it is needed within the ibis project.

## Context / Background
* I am working on different `exasol` integrations/projects and therfore I have a need to be flexible regarding the configuration
   * From a single project perspective (ibis) with "a known set of settings" it should not be an issue to hard code this settings. (So I think I just took the wrong perspective on this one)
* The MYSQL and Postgres tests configurations support hard coded settings which can be overwritten via `env`
  * Because I already had some snippets to support `pytest`s configuration mechanism, I was wondering why not use it in the first place.

Reply:
I will replace it with the pattern in the `mysql` and `postgress` tests (hard coded + env).